bin/check-znode.rb

```diff
@@ -49,14 +50,26 @@ class CheckZnode < Sensu::Plugin::Check::CLI
          short: '-v',
          long: '--check_value REGEX'
 
+  option :check_child,
+         description: 'Optionally check sub-nodes against a regex',
+         short: '-c',
+         long: '--check_child REGEX'
+  
   def znode_status
     zk = Zookeeper.new(config[:server])
     znode = zk.get(path: config[:znode])
+    children = config[:check_child] ? "#{zk.get_children(path: config[:znode])[:children]}" : null
 
     if znode[:stat].exists?
       if config[:check_value]
         if Regexp.new(config[:check_value]).match(znode[:data])
           ok "#{config[:znode]} value matched regexp '#{config[:check_value]}'"
+        elsif config[:check_child]
```

Comment:
while it does lead to some duplication perhaps we allow checking the parent and the child objects by checking as this maintains backwards compatibility. It might make sense to split that out into its own function to help with making the logic a bit cleaner as there is already a lot of nested conditionals underneath the changed code.
```ruby
if config[:check_value] && config[:check_child]
  # stuff
elsif config[:check_value]
  # stuff
elsif config[:check_child]
  # stuff
else
  # crit stuff
end
...
```

Reply:
So instead adding the functionality to check_znode.py would your suggestion be to create a new check_znode_children.py?

Reply:
we could either add the functionality in a new check or keep the ability to do both as I outlined above. Maybe I am missing something but it seems like we can have our cake and eat it too here but I could be wrong.

Reply:
Sorry for the delay, was away for a few weeks.  I'm open to whatever is the best way to implement.  It's just functionality that we need.  I'm sure others would also find it useful.  What would you recommend?